src/editor/panels/TopPanel.js

```diff
@@ -422,6 +422,22 @@ class TopPanel {
 
       // if (elem)
     } else if (this.multiselected) {
+      // Check if all selected elements have the same tag
+      const selElems = this.editor.svgCanvas.getSelectedElements()
+      const tagNames = []
+      selElems.forEach(elem => {
```

Comment:
while it may not be a performance or memory issue, do we have:
1. keep the tag names in the array tagNames?
2. do this loop even if tagNames[0] is not 'text"

I would suggest to use every() on selElems and to check if they are all 'text'. 
every would stop on the first tag not 'text'

Reply:
Thanks for the suggestion, much cleaner this way!